### PR TITLE
Fetch data from /servers

### PR DESCRIPTION
**Pull request by karlenDimla**
Task/Issue URL: https://app.asana.com/0/0/1205715197029706/f

### Description
See attached task description

### Steps to test this PR
See attached task description

### CODE REVIEW

**Comment by karlenDimla** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3650** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3650" target="_blank">Graphite</a> 
    * **PR #3673** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3673" target="_blank">Graphite</a> 
      * **PR #3674** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3674" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3674?utm_source=stack-comment).

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/geoswitching/GeoSwitchingContentProvider.kt [hunk 1]
```diff
@@ -16,52 +16,80 @@
...
-            ),
+@ContributesBinding(AppScope::class)
+class RealGeoSwitchingContentProvider @Inject constructor(
+    private val wgVpnControllerService: WgVpnControllerService,
+    private val dispatcherProvider: DispatcherProvider,
+    private val netPGeoswitchingRepository: NetPGeoswitchingRepository,
+) : GeoSwitchingContentProvider {
+    override suspend fun downloadData() {
+        withContext(dispatcherProvider.io()) {
+            getContent().map {
```
Comment (aitorvs) [inline network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/geoswitching/GeoSwitchingContentProvider.kt:48]: nit: rename to `getAvailableCountries`? 
getContent is very generic name

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/geoswitching/GeoSwitchingContentProvider.kt [hunk 2]
```diff
@@ -16,52 +16,80 @@
...
+import com.duckduckgo.di.scopes.AppScope
+import com.duckduckgo.networkprotection.impl.configuration.WgVpnControllerService
 import com.duckduckgo.networkprotection.impl.settings.geoswitching.GeoSwitchingContentProvider.AvailableCountry
+import com.duckduckgo.networkprotection.store.NetPGeoswitchingRepository
+import com.duckduckgo.networkprotection.store.db.NetPGeoswitchingLocation
 import com.squareup.anvil.annotations.ContributesBinding
 import javax.inject.Inject
+import kotlinx.coroutines.withContext
 
 interface GeoSwitchingContentProvider {
```
Comment (aitorvs) [inline network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/geoswitching/GeoSwitchingContentProvider.kt:29]: About naming, this is not really about geo switching but rather about getting the available VPN servers.
Should we rename to `VpnEgressServersProvider` or something like that?

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/geoswitching/GeoSwitchingContentProvider.kt [hunk 3]
```diff
@@ -16,52 +16,80 @@
...
-                cities = emptyList(),
-            ),
+                countryCode = it.countryCode,
+                countryName = it.countryName,
+                cities = it.cities,
+            )
+        }
+    }
+
+    private suspend fun getContent(): List<AvailableCountry> {
```
Comment (aitorvs) [inline network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/geoswitching/GeoSwitchingContentProvider.kt:70]: Similar naming comment, getContent is very generic and `AvailableCountry` is also more like `ServerLocation` ?

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/geoswitching/NetpGeoSwitchingViewModel.kt [hunk 4]
```diff
@@ -57,11 +57,11 @@ class NetpGeoSwitchingViewModel @Inject constructor(
     override fun onStart(owner: LifecycleOwner) {
         super.onStart(owner)
         viewModelScope.launch(dispatcherProvider.io()) {
-            val countryItems = contentProvider.getContent().map {
+            val countryItems = contentProvider.getDownloadedData().map {
```
Comment (aitorvs) [inline network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/geoswitching/NetpGeoSwitchingViewModel.kt:60]: Ditto about naming

**Review by aitorvs — APPROVED** [review]
LGTM. I have some comments about naming, I think they're not really clear.
I think in general all methods are in relation to server locations rather than geoswitching. How we use those locations maybe ge-switching or something else.

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/geoswitching/GeoSwitchingContentProvider.kt (hunk 1 above)
Comment (karlenDimla) [reply network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/geoswitching/GeoSwitchingContentProvider.kt]: Updated `downloadData` to `downloadServerLocations`

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/geoswitching/GeoSwitchingContentProvider.kt (hunk 2 above)
Comment (karlenDimla) [reply network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/geoswitching/GeoSwitchingContentProvider.kt]: Renamed to `NetpEgressServersProvider`

File: network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/geoswitching/GeoSwitchingContentProvider.kt (hunk 3 above)
Comment (karlenDimla) [reply network-protection/network-protection-impl/src/main/java/com/duckduckgo/networkprotection/impl/settings/geoswitching/GeoSwitchingContentProvider.kt]: Updated `AvailableCountry` to `ServerLocation`

**Pull request merged**